src/agents/tracing/processors.py

```diff
@@ -70,10 +72,24 @@ def set_api_key(self, api_key: str):
         """
         self.api_key = api_key
 
+    def _lazy_read_api_data_from_env(self):
+        if self._lazy_read_complete:
+            return
+
+        self._lazy_read_complete = True
+        if not self.api_key:
+            self.api_key = os.environ.get("OPENAI_API_KEY")
+        if not self.organization:
+            self.organization = os.environ.get("OPENAI_ORG_ID")
+        if not self.project:
+            self.project = os.environ.get("OPENAI_PROJECT_ID")
```

Comment:
consider using built-in functools.cached_property from python 3.8 rather than managing state yourself:

```suggestion
    @cached_property
    def api_key(self):
        return self._api_key or os.environ.get("OPENAI_API_KEY")
    
    @cached_property
    def organization(self):
        return self._organization or os.environ.get("OPENAI_ORG_ID")

    @cached_property
    def project(self):
        return self._project or os.environ.get("OPENAI_PROJECT_ID")
```